lib/classes/Variables.js

```diff
@@ -145,7 +148,7 @@ class Variables {
   }
 
   getValueFromSelf(variableString) {
-    let valueToPopulate = _.cloneDeep(this.service);
+    let valueToPopulate = this.service;
```

Comment:
I thought this was the whole point!  ;)

Actually, changing this moved me closer to enabling my scenario but may be legacy no longer necessary because of other changes.  I have to admit that I haven't completely groked this code.

What I observed was that the attribute value ${self:} was being copied and then populate was being called again.  I was concerned that it would infinitely recurse on that.  I saw evidence later that expansion is on demand so I will try and remove this today and test it out some more.